### DISCUSSION TITLE
Sending BLE data with a GET request.

### DISCUSSION THREAD

**Discussion by cyrax72** [arendst/Tasmota#22221] [Q&A] [answered]
Hello. I have BLE sensors that send strings like 
23:03:33.471 RSL: SENSOR = {"Time":"2024-09-29T23:03:33","ATC89617e":{"mac":"a4c13889617e","Temperature":24.5,"Humidity":48.8,"DewPoint":13.0,"Btn":0,"Battery":100,"RSSI":-68},"TempUnit":"C"}
The names (here ATC89617e) are different for everyone. 
I need to send each SENSOR message to another device with a GET request. 
I tried to do this using Rule, but I can't figure out how to send everything at once, with any sensor name (the whole string, or by individual parameters). 
Please tell me how to do it better?

**Comment by sfromis** [comment]
Simplest would be to use MQTT, and have the other device subscribe to the payloads.

Using  rules, there is no way of getting the full sensor payloads; closest is that you can have separate rules to pick up each field one by one, and then build a message again at last.

BLE should mean that your Tasmota device is ESP32-based, where the Berry language provides greatly enhanced flexibility instead of rules, including picking up the full sensor payload, and then you can do whatever, including `http` requests.

**Reply by cyrax72** [reply]
Yes, I was thinking about Berry. but I'm not a programmer.. I read the list of Berry commands, but didn't find any related to BLE... maybe you came across some examples of similar scripts?

**Reply by sfromis** [reply, marked as answer]
What you asked about did not relate to anything BLE-specific in Berry, but instead about the SENSOR message, which you wanted to send to another device via http. It should make no difference if this "happens" to be data originating from BLE.

Using `tasmota.add_rule`, you could pick up the sensor message, all data included, and use `webclient` to make http requests to the other machine. Something like this (untested):
```
tasmota.add_rule('ATC89617e',
  def (value, trigger, payload, bin)
    import json
    var data = json.dump(value)  # sensor fields - or payload for full message
    var wc = webclient()
    wc.begin('http://myserver/endpoint?data=' + data)
    var status = wc.GET()
    var response = wc.get_string()
    print(status, response)
    wc.close()
  end)
```

**Reply by cyrax72** [reply]
Thank you again. I managed to completely solve the problem, if anyone is interested, here is the final script. It sends a GET request with the full JSON received from BLE sensors, for different sensors it is enough to simply add triggers with their MAC addresses:

```
def rule_blesend (value, trigger, payload, bin)
    import json
    var data = json.dump(value)       
    var parsed_data = json.load(data) 
    var macaddr = parsed_data['mac'] 
    var wc = webclient()
    var url = "http://192.168.11.7/api/obj?name=" + macaddr + "&value=" + data; 
    wc.begin(url) 
    var status = wc.GET()
    var response = wc.get_string()
    print(status, response)   
    wc.close()
end

tasmota.add_rule("ATC89617e",rule_blesend)
tasmota.add_rule("ATC012fda",rule_blesend)
```
